torchao/quantization/quant_api.py

```diff
@@ -214,3 +221,64 @@ def replace_conv2d_1x1(conv):
     _replace_with_custom_fn_if_matches_filter(
         model, replace_conv2d_1x1, filter_fn=filter_fn
     )
+
+
+def _register_parametrizations(model, subclass_constructors, filter_fn=None):
+    def insert_parametrization(lin):
+        for subclass_constructor in subclass_constructors:
+            _, args = lin.weight.__tensor_flatten__()
+            parametrize.register_parametrization(lin, "weight", subclass_constructor(*args))
+
+    _replace_with_custom_fn_if_matches_filter(
+        model,
+        insert_parametrization,
+        _is_linear if filter_fn is None else filter_fn,
+    )
+
+    return model
+
+def _get_linear_subclass_inserter(constructor, **kwargs):
+    def insert_subclass(lin):
+        # so that we don't modify the original kwargs
+        copied_kwargs = dict(kwargs)
+        get_block_size = copied_kwargs.pop("get_block_size", None)
+        if get_block_size:
+            block_size = get_block_size(lin.weight)
+            copied_kwargs["block_size"] = block_size
+        lin.weight = torch.nn.Parameter(constructor(lin.weight, **copied_kwargs), requires_grad=False)
+        return lin
+
+    return insert_subclass
+
+class WeightQuantizer(Quantizer):
+    def __init__(self, **kwargs):
+        super().__init__()
+        self.kwargs = kwargs
+
+    def quantize(self, model: torch.nn.Module, filter_fn=None) -> torch.nn.Module:
+        _replace_with_custom_fn_if_matches_filter(
+            model,
+            _get_linear_subclass_inserter(to_aqt, **self.kwargs),
```

Comment:
Is `to_aqt` also something that could become a parameter of `WeightQuantizer`?

Then you're really just using this to change the dtype of the weight of all nn.Linear layers. 

In that case, would this code be equivalent to

```
def quantize(model, filter_fn, **kwargs):
    for name, mod in model.named_modules():
        if isinstance(mod, torch.nn.Linear):
            mod.weight = torch.nn.Parameter(to_aqt(mod.weight, **kwargs))
```

or maybe even

```
def fn(mod):
    return to_aqt(mod, **kwargs)

def quantize(model, filter_fn, fn):
    for name, mod in model.named_modules():
        if isinstance(mod, torch.nn.Linear):
            mod.weight = torch.nn.Parameter(fn(mod.weight))
```

Reply:
yeah it feels like we can further unify `WeightQuantizer` and `DynamicActQuantizer` actually, I'll call it a `TensorSubclassQuantizer` for now